.github/workflows/dev-green.yml

```diff
@@ -0,0 +1,47 @@
+name: Deploy to Dev-green
+
+on: workflow_dispatch
+
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    environment: dev-green
+    concurrency: dev-green
+    env:
+      AWS_REGION: us-east-1
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get commit metadata
+        run: |
+          echo SHA_SHORT=$(git rev-parse --short ${{ github.sha }}) >> $GITHUB_ENV
+          echo COMMIT_MSG=$(git log --format=%B -n 1 ${{ github.sha }} | head -n 1) >> $GITHUB_ENV
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+      - name: Login to Amazon ECR
+        uses: aws-actions/amazon-ecr-login@v1
+      - run: ./build.sh
+        env:
+          APP: api
+          DOCKER_REPO: ${{ secrets.DOCKER_REPO }}
+      - name: Deploy to EB
+        uses: einaregilsson/beanstalk-deploy@v17
```

Comment:
I'm not quite sure how the version pinning works. Can `@v17` be updated out from under us? After the codecov incident, I'm sort of gunshy about using 3rd party scripts like this which look like they have access to AWS keys. I'm squinting and it looks similar to downloading the codecov script and running it, but maybe the `@v17` is akin to pinning to a checksum?

If not, I wonder if it would be worthwhile forking that action, and then using our own fork.

The same applies to `aws-actions`, really, but that feels a little more trustworthy. But I'm open to a discussion about having a rule of forking and using only our own actions. (Which would then entail a fair bit of work keeping them up to date, etc.)

Reply:
It can; that's a good point. I'll create a fork.